### DISCUSSION TITLE
Mirth Connect Administrator Launcher 1.4.1 hangs on 4.0 servers, not on 3.5.

### DISCUSSION THREAD

**Discussion by jwhowa-rp** [nextgenhealthcare/connect#5928] [General]
Greetings everyone,

Since I upgraded to Mirth Connect Administrator Launcher 1.4.1 and utilized the default Java installation, I've observed some irregularities with the launcher:

- When logging into a version 3.5 Mirth server, the process is smooth and everything functions as expected.
- However, for a 4.0.1 server, here's the sequence of events: The login screen appears, I enter my credentials, then post the Java Virtual Environment processes, a blank "mirth" screen displays. This blank screen persists indefinitely. Interestingly, the moment I close the launcher, the Mirth interface appears as expected.

I handle multiple servers across both these versions. The version 3.5 servers operate without any glitches, even when I switch between servers or manage multiple at once.

On the other hand, every time I attempt to log into a version 4.0 server, the launcher stalls. This forces me to repetitively close the launcher, select another system, and restart the process.

Has anyone encountered this? Any insights into what might be causing this behavior?

**Comment by jonbartels** [comment]
Smells like https://github.com/nextgenhealthcare/connect/issues/4333 and https://github.com/nextgenhealthcare/connect/issues/5765

When you launch MC turn on "Show Java Console" and share the output. 

If you stop _the launcher_ after getting the blank screen does the Mirth admin tool start up cleanly?

You can also try https://github.com/kayyagari/ballista . This is a fully open source launcher tool that replaces MCAL. It is known to work well on 4.x versions of MC. Should work on 3.x but not much data yet.

**Comment by jwhowa-rp** [comment]
> If you stop the launcher after getting the blank screen does the Mirth admin tool start up cleanly?

Sure does.

In fact, I can use MCAL and select several systems to start loading.  As their login banner pops up, I can log in, and eventually all servers will be loaded with blank (white) screens.  As soon as I close MCAL, all those white screens IMMEDIAETLY go to the dashboard on all servers.  Blink of an eye fast.

It is clearly already loaded all that in the background, just not going to show it to me until I close the MCAL.

If I use the same version of MCAL and log in to 3.x Mirth servers, this does not happen.  Only when logging into 4.0.1 servers.

I have tried with and without the legacy settings.  I have used Java 8 and 17.

My java log looks normal, except it just "pauses" until I close the MCAL.

Any other ideas?

**Reply by jonbartels** [reply]
Thats a fault in the Launcher. The root cause appears to be the launcher doing something with the `stdout` stream. MCA tries to write something to that stream but MCAL has not properly flushed the stream. Since MCA can't write to `stdout` it just hangs. The details are in the issues I linked earlier.

What OS are you on? The issue is most common on Windows but has been reported on MacOS too.

You have two options:
* Try https://github.com/kayyagari/ballista - This is a fully open source replacement for MCAL. The disadvantage is that it does not bundle a JDK, you have to manage that yourself
* Use the "Close this window after launching" option - It will close MCAL as soon as MCA starts. This is annoying because you'll have to start MCAL each time you need to open a new MCA

NextGen is aware of the issue but I'm not aware of their workload or release cycle.
